### DISCUSSION TITLE
Navigation to invoke function per a specific route

### DISCUSSION THREAD

**Discussion by andrew-tud** [SAP/luigi#3019] [Q&A] [answered]
Is it possible in Luigi Framework to specify a routing that will call a JavaScript function when the user invoked the routing?

One of the things I am trying to do is to hide the top and side navigations of the MFE shell if the user visited a particular URL path. It would be nice if this can be done by calling a function that will change this config:
```
 settings: {
      appLoadingIndicator: {
        hideAutomatically: true
      },
      hideNavigation: false
  }
```

to set hideNavigation to true before loading the path to make the MFE architecture vertically split on runtime.

I tried various means of doing this (calling function when user invoked a routing) but I was unable to do.

**Comment by JohannesDoberer** [comment, marked as answer]
Hello,
yes, it is possible. You can use the property [nodeChangeHook](https://docs.luigi-project.io/docs/navigation-parameters-reference/?section=nodechangehook). 
It gets two parameters:
```
navigation:{
   nodeChangeHook: (prevNode, nextNode) => {
     //do something
   },
   nodes:[],
}
```
